src/main/java/io/cdap/plugin/spark/dynamic/ScalaSparkCompute.java

```diff
@@ -110,6 +110,11 @@ public void prepareRun(SparkPluginContext context) throws Exception {
       // If there is no output schema configured, derive it from the DataFrame
       // Otherwise, assume the DataFrame has the correct schema already
       outputSchema = DataFrames.toSchema((DataType) invokeDataFrameMethod(result, "schema"));
+    }else{
+      Schema dataSchema = DataFrames.toSchema((DataType) invokeDataFrameMethod(result, "schema"));
+      if(dataSchema.getFields().size() < outputSchema.getFields().size()){
+        throw new Exception("Invalid schema. Output schema is not matching input schema.");
```

Comment:
if this check is happening at validation time, why are we not using failure collector?

Reply:
Fixed